transformers/storage/vat/repository_test.go

```diff
@@ -756,17 +770,69 @@ var _ = Describe("Vat storage repository", func() {
 
 				It("deletes urn state associated with diff if it's the earliest state in the table", func() {
 					initialArt := rand.Int()
-					setupErrOne := repo.Create(diffID, headerOne.Id, urnArtMetadata, strconv.Itoa(initialArt))
+					setupErrOne := repo.Create(diffIdOne, headerOne.Id, urnArtMetadata, strconv.Itoa(initialArt))
 					Expect(setupErrOne).NotTo(HaveOccurred())
 
-					_, deleteErr := db.Exec(deleteRowQuery, headerOne.Id)
+					_, deleteErr := db.Exec(deleteHeaderQuery, headerOne.Id)
 					Expect(deleteErr).NotTo(HaveOccurred())
 
 					var urnStates []test_helpers.UrnState
 					queryErr := db.Select(&urnStates, getArtQuery)
 					Expect(queryErr).NotTo(HaveOccurred())
 					Expect(len(urnStates)).To(Equal(0))
 				})
+
+				It("handles removing snapshots when there's a three-part reorg", func() {
+					// this test case mimics the issue that is detailed in VDB-1756 where we saw a reorg that resulted in
+					// receiving three headers - canonical, noncanonical, canonical
```

Comment:
Seems like this test is saying we get:
- canonical for first block
- canonical for second block
- non-canonical for second block
Is that right? The comment makes me think the test would deal with everything at the same header, but then I don't understand incrementing the block number on line 803 🤔

Reply:
Yeah, this assumption is correct. I added a "before" block, so that we'd make sure to have some data in the db before testing this scenario out to make sure that we weren't unintentionally manipulating snapshots that were already in the db, and shouldn't have been changed. Going to add some additional assertions and comments to hopefully clear up that confusion some.

Reply:
I'm totally on board with including the before block to verify we're only manipulating the snapshot in question 👍